### PR TITLE
[VC-43] Acceptance criteria silently dropped when it appears at end of description

### PR DESCRIPTION
**Pull request by cedricfarinazzo**
## VC-43 — Acceptance criteria silently dropped when it appears at end of description

**Jira ticket:** https://sedinfra.atlassian.net/browse/VC-43

### Description

extractAcceptanceCriteria in tickets.go:221-224 searches for a newline after the "Acceptance Criteria" heading. If the heading is the last line of the description with no trailing newline, strings.Index returns -1 and the function returns "" — the entire acceptance criteria section is discarded with no error or warning. This causes the agent to proceed without acceptance criteria, silently reducing implementation quality.\n\nFile: internal/jira/tickets.go:221-224\n\nFix: When no newline is found after the heading, treat the remainder of the string as the acceptance criteria body rather than returning empty.

---
*Generated by [Nightshift](https://github.com/cedricfarinazzo/nightshift) — automated agent*

### CODE REVIEW

**Comment by cedricfarinazzo** [comment]
🤖 **Nightshift — Implementation Summary**

**Ticket:** [VC-43 — Acceptance criteria silently dropped when it appears at end of description](https://sedinfra.atlassian.net/browse/VC-43)

### What was done

Now I have a clear picture. Let me implement the fix:

Now update the existing test comment and add two new tests:

All 25 packages pass. The fix is complete:

- **`internal/jira/tickets.go`**: When no `\n` follows the "acceptance criteria" heading, the function now returns the inline content after the keyword (stripping `": "`) instead of `""`.
- **`internal/jira/tickets_test.go`**: Updated the existing test comment; added two new tests — one for the bug case (inline content, no newline) and one regression guard (normal heading + items, no trailing newline).



---
*Generated automatically by [Nightshift](https://github.com/cedricfarinazzo/nightshift)*

**Comment by cedricfarinazzo** [comment]
🤖 **Nightshift — Review Feedback Addressed**

I've addressed the reviewer feedback.

### Summary

I’ve patched the extraction logic and added the regression tests; now I’m formatting and running the targeted tests to catch any edge cases.

Fixed. `extractAcceptanceCriteria` now only matches real heading lines, preserves inline text on the heading line even when more content follows on later lines, and ignores plain sentence mentions. I also added regression tests for the inline-with-newline case and the false-positive mention case.



_This update was generated automatically by Nightshift._

File: internal/jira/tickets.go [hunk 1]
```diff
@@ -303,31 +302,72 @@ func isBlockNode(n *model.CommentNodeScheme) bool {
...
-	start += idx + 1
-	rest := description[start:]
-	// Stop at the next section: a non-empty line that ends with ':' and contains no spaces
-	// (matches "SectionName:" style headings produced by extractText).
-	lines := strings.Split(rest, "\n")
-	var out []string
-	for _, line := range lines {
-		trimmed := strings.TrimSpace(line)
-		if len(out) > 0 && trimmed != "" && strings.HasSuffix(trimmed, ":") && !strings.Contains(trimmed, " ") {
+	const acKeyword = "acceptance criteria"
```
Comment (cedricfarinazzo) [inline internal/jira/tickets.go]: run golangci-lint
  Running [/home/runner/golangci-lint-1.64.8-linux-amd64/golangci-lint config path] in [/home/runner/work/nightshift/nightshift] ...
  Running [/home/runner/golangci-lint-1.64.8-linux-amd64/golangci-lint run] in [/home/runner/work/nightshift/nightshift] ...
  Error: internal/jira/tickets.go:308:8: const `acKeyword` is unused (unused)
  	const acKeyword = "acceptance criteria"
  	      ^
  
  Error: issues found
  Ran golangci-lint in 4810ms
